HealthcareApp/src/Healthcare.Infrastructure/Repository/Dapper/MetricsRepository.cs

```diff
@@ -0,0 +1,76 @@
+using System.Text.RegularExpressions;
+using Application.Abstractions.Dapper;
+using Dapper;
+using Healthcare.Infrastructure.Persistance;
+
+namespace Infrastructure.Repository.Dapper;
+
+public class MetricsRepository : IMetricsRepository
+{
+    private readonly SqlConnectionFactory _dbConnectionFactory;
+
+    public MetricsRepository(SqlConnectionFactory dbConnectionFactory)
+    {
+        _dbConnectionFactory = dbConnectionFactory;
+    }
+
+    public async Task<long> GetUsersCountByRoleAsync(int roleId)
+    {
+        using var sqlConnection = _dbConnectionFactory.CreateDbConnection();
+        const string query = """
+                             SELECT COUNT(*)
+                             FROM AspNetUserRoles
+                             WHERE RoleId = @RoleId
+                             GROUP BY RoleId
+                             """;
+
+        return await sqlConnection.QueryFirstOrDefaultAsync<long>(query, new { RoleId = roleId });
+    }
+
+    public async Task<List<object>> GetDoctorsWithAppointments()
```

Comment:
use DTO and look at my project, I created files for queries

Reply:
I agree that you can create model to map to, and dapper will map the fields to your model.

Note that for querying your model names should have the same name as column names in your result set

Reply:
Let's better change this query to `GetDoctorsAppointmentsCount` with timerange filter because it's not related to analytics or statistics now.

Its result should return count of doctors appointments which they had during the mentioned timerange